File: 3606-minimum-element-after-replacement-with-digit-sum/3606-minimum-element-after-replacement-with-digit-sum.py
class Solution:
    def minElement(self, nums: List[int]) -> int:
        SumList = [] #The final list I want to put the sum of the digits of the elements
        for i in nums: #accessing each elements of the list, to add the lists
            Elements = list(str(i)) #chaning eacch elements to list to add the digits
            SumOfDigits = 0 
            for j in Elements: #accessing each digits in each element
                SumOfDigits += int(j) #adding the digits
            SumList += [SumOfDigits] #concatinating each sum to create a list of sum of each elements

        return min(SumList)

        

        

File: 3606-minimum-element-after-replacement-with-digit-sum/3606-minimum-element-after-replacement-with-digit-sum.cpp
class Solution {
public:
    int sum(int n){
        int digit = n % 10; 
        if(n == 0){
            return 0; 
        }
        int Sum = digit + sum(n/10); 
        return Sum; 
    }
    int minElement(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++){
            nums[i] = sum(nums[i]); 
        }
        int Min = nums[0]; 
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] < Min){Min = nums[i];}
        }
        return Min; 
        
    }
};